client/nginx_test.go

```diff
@@ -224,9 +224,85 @@ func TestStreamDetermineUpdates(t *testing.T) {
 	}}
 
 	for _, test := range tests {
-		toAdd, toDelete := determineStreamUpdates(test.updated, test.nginx)
+		toAdd, toDelete, _ := determineStreamUpdates(test.updated, test.nginx)
 		if !reflect.DeepEqual(toAdd, test.expectedToAdd) || !reflect.DeepEqual(toDelete, test.expectedToDelete) {
 			t.Errorf("determiteUpdates(%v, %v) = (%v, %v)", test.updated, test.nginx, toAdd, toDelete)
 		}
 	}
 }
+
+func TestDetermineUpdatesWithUpdate(t *testing.T) {
+	maxConns := 1
+	var tests = []struct {
+		updated          []UpstreamServer
+		nginx            []UpstreamServer
+		expectedToUpdate UpstreamServer
+	}{{
+		updated: []UpstreamServer{
+			{
+				Server:   "10.0.0.1:80",
+				MaxConns: &maxConns,
+			},
+		},
+		nginx: []UpstreamServer{
+			{
+				Server: "10.0.0.1:80",
+			},
+			{
+				ID:     2,
+				Server: "10.0.0.2:80",
+			},
+		},
+		expectedToUpdate: UpstreamServer{
+			Server:   "10.0.0.1:80",
+			MaxConns: &maxConns,
+		}},}
+
+	for _, test := range tests {
+		_, _, toUpdate := determineUpdates(test.updated, test.nginx)
+
+		for _, update := range toUpdate {
```

Comment:
there is no need to loop through a slice here. the expected output of the function with regard to updates is a slice. We can compare slides  using `reflect.DeepEqual`.

Reply:
Yeah I think so, this was old code